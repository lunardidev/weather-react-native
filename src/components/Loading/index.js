import React from 'react';
import { ActivityIndicator } from 'react-native';

import { Container } from './styles';

export default function Loading() {
  return (
    <Container>
      <ActivityIndicator size="large" animating={true} color="#c3c3c3" />
    </Container>
  );
}
