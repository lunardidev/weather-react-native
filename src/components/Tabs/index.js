import React, { useState, useEffect } from 'react';
import propTypes from 'prop-types';

import {
  Container,
  TabContainer,
  TabItem,
  TabTitleText,
  TabText,
  TabContainerText,
  TabSubText,
  TabSubTextContainer,
  TouchableArea,
} from './styles';

const Tabs = (props) => {
  const { backgroundColor, selectedColor, items, onPress } = props;

  const [itemsTab, setItemsTab] = useState([]);

  useEffect(() => {
    setItemsTab(items);
  }, [items]);

  const onPressItem = (item) => {
    onPress(item);

    const newState = itemsTab.map((i) => {
      const changedItem = i;
      changedItem.actived = changedItem.date === item.date;

      return changedItem;
    });

    setItemsTab(newState);
  };

  return (
    <Container>
      <TabContainer>
        {itemsTab.map((item) => (
          <TabItem
            actived={item.actived}
            key={item.date}
            backgroundColor={backgroundColor}
          >
            <TouchableArea
              activeOpacity={0.4}
              underlayColor={backgroundColor}
              onPress={() => onPressItem(item)}
            >
              <TabContainerText
                selectedColor={selectedColor}
                actived={item.actived}
              >
                <TabTitleText
                  selectedColor={selectedColor}
                  actived={item.actived}
                >
                  {item.weekday}
                </TabTitleText>
                <TabText selectedColor={selectedColor} actived={item.actived}>
                  {item.description}
                </TabText>
                <TabSubTextContainer>
                  <TabSubText actived={item.actived}>
                    {item.max}&deg;
                  </TabSubText>
                  <TabSubText actived={item.actived}>
                    {' | '}
                    {item.min}&deg;
                  </TabSubText>
                </TabSubTextContainer>
              </TabContainerText>
            </TouchableArea>
          </TabItem>
        ))}
      </TabContainer>
    </Container>
  );
};

Tabs.propTypes = {
  backgroundColor: propTypes.string,
  selectedColor: propTypes.string,
  items: propTypes.arrayOf(
    propTypes.shape({
      date: propTypes.string,
      weekday: propTypes.string,
      max: propTypes.number,
      min: propTypes.number,
      description: propTypes.string,
      condition: propTypes.string,
      actived: propTypes.boolean,
    })
  ),
  onPress: propTypes.func,
};

Tabs.defaultProps = {
  backgroundColor: '#fff',
  selectedColor: '#fff',
};

export default Tabs;
