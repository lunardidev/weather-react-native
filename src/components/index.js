export { default as Tabs } from './Tabs';
export { default as Loading } from './Loading';
