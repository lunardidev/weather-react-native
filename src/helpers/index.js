export { default as conditionSlug } from './conditionSlug';
export { default as today } from './today';
