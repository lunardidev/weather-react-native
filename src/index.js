import React from 'react';
import { StatusBar } from 'react-native';

import './config/ractotron';

import { Main } from './layouts/Weather';

import { ApiContextProvider } from './contexts/apiContext';

const App = () => (
  <ApiContextProvider>
    <StatusBar
      backgroundColor="transparent"
      translucent
      barStyle="light-content"
    />
    <Main />
  </ApiContextProvider>
);

export default App;
