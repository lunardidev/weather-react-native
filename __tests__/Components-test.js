import React from 'react';
import renderer from 'react-test-renderer';

import { Loading, Tabs } from '../src/components';

test('should render the Tab component correctly', () => {
  const element = renderer
    .create(
      <Tabs
        onPress={(obj) => jest.fn()}
        items={[
          {
            date: '03/2020',
            weekday: 'SÃ¡bado',
            max: 25,
            min: 10,
            description: 'Sol',
            condition: 'clear',
            actived: true,
          },
        ]}
        backgroundColor="#fff"
        selectedColor="#fff"
      />
    )
    .toJSON();
  expect(element).toMatchSnapshot();
});

test('should render the Loading component correctly', () => {
  const element = renderer.create(<Loading />).toJSON();
  expect(element).toMatchSnapshot();
});
