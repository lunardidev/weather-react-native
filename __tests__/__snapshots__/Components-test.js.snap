// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Loading component correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;

exports[`should render the Tab component correctly 1`] = `
<View
  style={
    Array [
      Object {},
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "paddingLeft": 0,
        "paddingRight": 0,
      }
    }
    horizontal={true}
    showsHorizontalScrollIndicator={false}
    style={
      Array [
        Object {},
      ]
    }
  >
    <View />
  </RCTScrollView>
</View>
`;
